packages/cache/src/Cache.ts

```diff
@@ -0,0 +1,204 @@
+import type { CacheKey, CacheOptions } from './types'
+import type { ExtractPartial } from './utility-types/ExtractPartial'
+import { hashCacheKey } from './utils'
+
+type Sync = (...args: unknown[]) => unknown
+
+export type AllStatusCached<TData, TCacheKey extends CacheKey = CacheKey> =
```

Comment:
```suggestion
export type CachedStatus<TData, TCacheKey extends CacheKey = CacheKey> =
```
Do I really need to use "All"? How about the name "CachedStatus"?

Reply:
CachedStatus but Cached['status'] is 'idle' | 'pending' | 'resolved' | 'rejected'

Reply:
No, it seems like I can't think of a better name. I think going with AllCachedStatus would be better.